### DISCUSSION TITLE
Frequent helm-controller notifications "failed to verify artifact: computed checksum"

### DISCUSSION THREAD

**Discussion by ArtunSubasi** [fluxcd/flux2#2470] [Q&A]
Hello,

after upgrading to Flux2 v0.26.0 I started getting recurring warning events from the helm-controller (over the notification-controller and Slack) in the following form:

```
helmrelease/my-app.my-namespace
failed to verify artifact: computed checksum 'b27f1279ed6fa4253086d3debf6eed607d891734861390e2f6aa23067c055f3a' 
doesn't match advertised '7c3209a4850ecd3e091f2fc16dc5512a5c04f0acfc933c5fd93b725c10658bb3'
````

These happen multiple times each day, sometimes in the middle of the night although no changes are made to the helm charts or releases. I couldn't find any more information in the logs.

- I would really love to hear about ideas. Why can this be happening?
- Any debugging ideas?
- Is there any documentation about the checksum verification feature of Flux? I couldn't find anything except for a couple of hash method changes in the change logs

## Versions:

- K8s (AWS EKS): 1.21.5
- Flux: v0.26.0
- helm-controller: v0.16.0
- notification-controller: v0.21.0
- source-controller: v0.21.1

**Comment by bpinter** [comment]
I'm also in the same boat. 

`failed to verify artifact: computed checksum 'd6ff5877fe5d2bcaed8cde7db6e71104ac77368f8844730efc3b48e5f3d7401d' doesn't match advertised '93a0b29d6e6e16d8321825e04645b3ec083004f0650c934d14a7e42598799333'`

**Versions:**
K8S (VMware Tanzu): v1.20.7
flux: v0.27.2
helm-controller: v0.17.1
image-automation-controller: v0.20.0
image-reflector-controller: v0.16.0
kustomize-controller: v0.21.1
notification-controller: v0.22.2
source-controller: v0.21.2

**Comment by ArtunSubasi** [comment]
Relevant line in the helm-controller for analysis purposes: https://github.com/fluxcd/helm-controller/blob/5a293d2dc4b2e0311ec3f4de049ad3e5b99159f5/controllers/helmrelease_controller_chart.go#L162

**Comment by yznima** [comment]
Same here. Is there a way to ignore this specific alert?

**Reply by stefanprodan** [reply]
Yes, you can add a regex that matches `failed to verify artifact` to the exclusion list: https://fluxcd.io/docs/components/notification/alert/#example

**Reply by ArtunSubasi** [reply]
Great tip Stefan, thanks! That will suspend the notifications, reduce the number of warnings in total and increase the awareness for other warnings that may appear.

Logically it does not solve the cause. The checksum validation remains unusable. Since this seems to be an issue across teams and different k8s types, it would be interesting to figure out why does happen. Do you have any ideas about how to analyse the issue?

**Reply by hiddeco** [reply]
See https://github.com/fluxcd/source-controller/issues/610
